### PR TITLE
Fix tfds/image_classification bugs on windows

### PR DESCRIPTION
**Pull request by vijayphoenix**
Related to https://github.com/tensorflow/datasets/issues/1911#issuecomment-616220757

### CODE REVIEW

**Comment by vijayphoenix** [comment]
New results
```
============================= test session starts =============================
platform win32 -- Python 3.7.6, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
rootdir: C:\Users\VIJAY\Desktop\GitHub_Repos\datasets
plugins: forked-1.1.3, xdist-1.31.0
collected 355 items

tensorflow_datasets\image_classification\beans_test.py ....s             [  1%]
tensorflow_datasets\image_classification\bigearthnet_test.py ....s       [  2%]
tensorflow_datasets\image_classification\binary_alpha_digits_test.py ... [  3%]
.s                                                                       [  4%]
tensorflow_datasets\image_classification\caltech_birds_test.py ....s.... [  6%]
s                                                                        [  7%]
tensorflow_datasets\image_classification\caltech_test.py ....s           [  8%]
tensorflow_datasets\image_classification\cars196_test.py ....s           [  9%]
tensorflow_datasets\image_classification\cassava_test.py ....s           [ 11%]
tensorflow_datasets\image_classification\cats_vs_dogs_test.py ....s      [ 12%]
tensorflow_datasets\image_classification\cbis_ddsm_test.py ....s....s... [ 16%]
.s                                                                       [ 16%]
tensorflow_datasets\image_classification\chexpert_test.py ....s          [ 18%]
tensorflow_datasets\image_classification\cifar10_1_test.py ....s         [ 19%]
tensorflow_datasets\image_classification\cifar10_corrupted_test.py ....s [ 21%]
                                                                         [ 21%]
tensorflow_datasets\image_classification\cifar_test.py ....s....s        [ 23%]
tensorflow_datasets\image_classification\citrus_test.py .F..s            [ 25%]
tensorflow_datasets\image_classification\cmaterdb_test.py ....s          [ 26%]
tensorflow_datasets\image_classification\colorectal_histology_test.py .. [ 27%]
..s....s                                                                 [ 29%]
tensorflow_datasets\image_classification\cycle_gan_test.py ....s         [ 30%]
tensorflow_datasets\image_classification\deep_weeds_test.py ....s        [ 32%]
tensorflow_datasets\image_classification\diabetic_retinopathy_detection_test.py . [ 32%]
...s                                                                     [ 33%]
tensorflow_datasets\image_classification\dmlab_test.py ....s             [ 35%]
tensorflow_datasets\image_classification\dtd_test.py ....s               [ 36%]
tensorflow_datasets\image_classification\eurosat_test.py ....s           [ 38%]
tensorflow_datasets\image_classification\flowers_test.py ....s           [ 39%]
tensorflow_datasets\image_classification\food101_test.py ....s           [ 40%]
tensorflow_datasets\image_classification\geirhos_conflict_stimuli_test.py . [ 41%]
...s                                                                     [ 42%]
tensorflow_datasets\image_classification\horses_or_humans_test.py ....s  [ 43%]
tensorflow_datasets\image_classification\image_folder_test.py ....s      [ 45%]
tensorflow_datasets\image_classification\imagenet2012_corrupted_test.py . [ 45%]
...s                                                                     [ 46%]
tensorflow_datasets\image_classification\imagenet2012_subset_test.py ... [ 47%]
.s                                                                       [ 47%]
tensorflow_datasets\image_classification\imagenet_resized_test.py ....s  [ 49%]
tensorflow_datasets\image_classification\imagenet_test.py ....s          [ 50%]
tensorflow_datasets\image_classification\imagenette_test.py ....s        [ 52%]
tensorflow_datasets\image_classification\imagewang_test.py ....s....s... [ 55%]
.s                                                                       [ 56%]
tensorflow_datasets\image_classification\inaturalist_test.py ....s       [ 57%]
tensorflow_datasets\image_classification\lfw_test.py ....s               [ 59%]
tensorflow_datasets\image_classification\malaria_test.py ....s           [ 60%]
tensorflow_datasets\image_classification\mnist_corrupted_test.py ....s   [ 61%]
tensorflow_datasets\image_classification\mnist_test.py ....s....s....s.. [ 66%]
..s                                                                      [ 67%]
tensorflow_datasets\image_classification\oxford_flowers102_test.py ....s [ 69%]
                                                                         [ 69%]
tensorflow_datasets\image_classification\oxford_iiit_pet_test.py ....s   [ 70%]
tensorflow_datasets\image_classification\patch_camelyon_test.py ....s    [ 71%]
tensorflow_datasets\image_classification\pet_finder_test.py ....s        [ 73%]
tensorflow_datasets\image_classification\places365_small_test.py ....s   [ 74%]
tensorflow_datasets\image_classification\plant_leaves_test.py ....s      [ 76%]
tensorflow_datasets\image_classification\plant_village_test.py .F..s     [ 77%]
tensorflow_datasets\image_classification\plantae_k_test.py ....s         [ 78%]
tensorflow_datasets\image_classification\quickdraw_test.py ....s         [ 80%]
tensorflow_datasets\image_classification\resisc45_test.py ....s          [ 81%]
tensorflow_datasets\image_classification\rock_paper_scissors_test.py ... [ 82%]
.s                                                                       [ 83%]
tensorflow_datasets\image_classification\smallnorb_test.py ....s         [ 84%]
tensorflow_datasets\image_classification\so2sat_test.py ....s            [ 85%]
tensorflow_datasets\image_classification\stanford_dogs_test.py ....s     [ 87%]
tensorflow_datasets\image_classification\stanford_online_products_test.py . [ 87%]
...s                                                                     [ 88%]
tensorflow_datasets\image_classification\stl10_test.py ....s             [ 90%]
tensorflow_datasets\image_classification\sun_test.py ....s....s          [ 92%]
tensorflow_datasets\image_classification\svhn_test.py ....s              [ 94%]
tensorflow_datasets\image_classification\uc_merced_test.py ....s         [ 95%]
tensorflow_datasets\image_classification\vgg_face2_test.py ....s         [ 97%]
tensorflow_datasets\image_classification\visual_domain_decathlon_test.py . [ 97%]
...s....s                                                                [100%]

================================== FAILURES ===================================
C:\ProgramData\Miniconda3\lib\site-packages\absl\testing\absltest.py:1704: AssertionError: [{'image': array([[[79, 79, 79],
C:\Users\VIJAY\Desktop\GitHub_Repos\datasets\tensorflow_datasets\core\tfrecords_writer.py:108: AssertionError: No examples were yielded.
=========================== short test summary info ===========================
FAILED tensorflow_datasets/image_classification/citrus_test.py::CitrusLeavesTest::test_download_and_prepare_as_dataset
FAILED tensorflow_datasets/image_classification/plant_village_test.py::PlantVillageTest::test_download_and_prepare_as_dataset 
===== 2 failed, 282 passed, 71 skipped, 597 warnings in 540.06s (0:09:00) =====
```

The remaining two errors are because of
* `tensorflow-io` errors on windows. 
* `tensorflow (tf.io.glob)` respectively.

So, we can deal with them in `tfds`

File: tensorflow_datasets/image_classification/places365_small.py [hunk 1]
```diff
@@ -134,6 +134,7 @@ def _generate_examples(self, archive, path_prefix, split_name,
         file_to_class = {x[0]: int(x[1]) for x in csv.reader(f, delimiter=" ")}
 
     for fname, fobj in archive:
+      fname = fname.replace("\\", "/")
       assert fname.startswith(path_prefix)
```
Comment (Conchylicultor) [inline tensorflow_datasets/image_classification/places365_small.py:138]: I'm confused how can the test pass if `path_prefix` is created with `os.path.join` (so contains the '\\'), but `fname.replace` contains the '/' ?

Comment (vijayphoenix) [reply tensorflow_datasets/image_classification/places365_small.py:138]: From what I understand,
`path_prefix` is a string constant like `test_256`, `val_256` and from the code, it does not look like it is created by `os.path.join`.

Comment (Conchylicultor) [reply tensorflow_datasets/image_classification/places365_small.py:138]: Yes, indeed, I misread the code. Sorry.

**Pull request merged**